### ISSUE TITLE
Reports that Riot 0.16 is harder accessibility-wise

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#7119]
From #riot: https://matrix.to/#/!DgvjtOljKujDBrxyHk:matrix.org/$1533103152976918nzEhb:matrix.org

> ... I find that there have been several changes to the interface in the last two releases that have made things quite difficult to use Riot via the browser or the desktop app.

> In the recent two releases, the ability to read the name of the room in the list of rooms is no longer possible. For example, I have to open each room to now find out which room I'm in. THis could present a significant problem when I have 60 or so personal chats and 25 or so general rooms.

> Previously, when I navigated to the names of the rooms, I was able to identify the room by its name or the person's name.

> In the most recent release, I am no longer able to focus on the edit field to enter text. This is even a bigger problem when trying to chat.

https://www.nvaccess.org/ was given as an example screen reader

**Comment by lampholder** [comment]
Thanks @turt2live @PratikP1 - sorry about this :(

@PratikP1 I've installed NVDA - can you describe the behaviour you _used_ to see (if that was ideal) and the behaviour you'd _like_ to see (if that's different)?

**Comment by PratikP1** [comment]
@lampholder wrote:
> can you describe the behaviour you used to see (if that was ideal) and the behaviour you'd like to see (if that's different)?


Since I'm working on a project and with a team that is under NDA, I'm unable to provide screenshots or live examples. However here's what is happening and what's expected. I'll provide the most critical things to fix here. We should have a separate ticket for what could make the experience excellent.
* currently, the user navigates to rooms/chats by pressing the letter b to move to and from button to button that corresponds to each room or DM. In current version of the desktop app and the browser, moving from button to button only announces "DM button" or "button." In previous versions, the announcement was something similar to "dmxxx button" where the xxx corresponded to the name of the user. This is with direct messages or individual chats. In other words, the buttons are no longer labeled with the user or room name. Only the control type is identified. 
* The "people" or "rooms" labels no longer exist. Only the number of unread items is identified. Ideally, it should read "people, unread 13."
* The "more" button has also lost its label.
* I can't describe what is happening with the edit field to input text without providing a little bit of background. Current screen readers use browser APIs such as Microsoft UIA or IAccessible2 to build a virtual buffer of the contents of browsers or browser based apps. To interact with general contents and most form elements, the user is placed in browse mode, which allows the user to use up/down or left/right keys to navigate within the content as if she were in a document. To interact with edit fields and list boxes, browse mode must be switched off. Pressing the enter key when focused on an edit field does this. This now allows the user to interact with the edit field and write within it. In the latest update, the user is no longer able to press the enter key on the text input field to switch browse mode off, effectively disabling the user's ability to type. The browse mode has to be swtiched off manually. This is not typical interaction and takes significantly longer. In addition, NVDA crashes or freezes when selecting all the text in the entire  field to delete or replace. The screen reader has to be restarted. In previous versions, the edit fields worked well.


These are the two most critical issues at the moment. Let me know if you need help understanding the screen reader and the various modalities.

**Comment by PratikP1** [comment]
Just a quick report to let you know that issues reported above appeared to have been fixed after the last update. However the labels disappeared all of a sudden yesterday. I am uncertain  as to why this could have occurred.

**Comment by t3chguy** [comment]
Closing as stale, all of the above have been either fixed or covered in other issues.
